data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -135,8 +149,10 @@ protected long pos(T record) {
       Iterable<CloseableIterable<Record>> deleteRecords = Iterables.transform(deletes,
           delete -> openDeletes(delete, deleteSchema));
       StructLikeSet deleteSet = Deletes.toEqualitySet(
-          // copy the delete records because they will be held in a set
-          CloseableIterable.transform(CloseableIterable.concat(deleteRecords), Record::copy),
+          CloseableIterable.transform(
+              CloseableIterable.combine(deleteRecords, readService, readParallelism),
+              deleteRecord -> (StructLike) deleteRecord
```

Comment:
Is the outer `transform` call needed? The iterable should be of `Record` which is already `StructLike`.